packages/peregrine/src/hooks/useDOMQuery.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect, useCallback } from 'react';
+
+const findDOMElements = queryString =>
+    document.querySelectorAll(queryString) || [];
+
+/**
+ * A custom hook that will return the array of elements that
+ * match the queryString. document.querySelectorAll is used
+ * to find all elements.
+ *
+ * @param {string} queryString
+ * @return Array<HTMLElement> and {
+ *  setInnerHTML: (string) => void,
+ *  setInnerText: (string) => void,
+ *  setAttribute: (string, any) => void
+ * }
+ */
+export const useDOMQuery = queryString => {
+    const [elements, setElements] = useState([]);
+    useEffect(() => {
+        setElements(findDOMElements(queryString));
+    }, []);
+    const setInnerHTML = useCallback(
+        content => {
+            elements.forEach(element => {
+                element.innerHTML = content || element.innerHTML;
```

Comment:
Do we need this `||` here? Why would anyone pass in falsy `content`?

If we do need it, consider 

```js
if (content) {
 // forEach here
}
```

It will save some time if we have a lot of `elements` - we can just skip the entire iteration since we won't be updating anything anyway.